packages/web/src/js/ScrollView.ts

```diff
@@ -0,0 +1,126 @@
+// The `ScrollView` component is currently untested. This is because it is not possible to test it in a headless
+// environment (tests fail on `getBoundingClientRect` that we need to mock somehow).
+
+import BaseComponent from './BaseComponent';
+import { EventHandler, SelectorEngine } from './dom';
+import { debounce, enableToggleAutoloader } from './utils';
+
+const NAME = 'scrollView';
+const DEFAULT_DIRECTION = 'vertical';
+const VIEWPORT_SELECTOR = '[data-spirit-element="viewport"]';
+const CONTENT_SELECTOR = '[data-spirit-element="content"]';
+const IS_SCROLLED_AT_START_CLASS_NAME = 'is-scrolled-at-start';
+const IS_SCROLLED_AT_END_CLASS_NAME = 'is-scrolled-at-end';
+const RESIZE_EVENT = 'resize';
+const SCROLL_EVENT = 'scroll';
+
+// Method `getElementsPositionDifference` sometimes returns floating point values that results in inaccurate detection
+// of start/end. It is necessary to accept this inaccuracy and take every value less or equal to 1px as start/end.
+const EDGE_DETECTION_INACCURACY_PX = 1;
+
+export type Direction = 'horizontal' | 'vertical';
```

Comment:
Please let it be known from the name that this is the type I propose to rename to `DirectionType` or `ScrollViewDirectionType`

Reply:
This is the topic to discuss. As we do not have any guidelines about the naming of the types and we do not use `Type` suffix anywhere in this package, I am ok with this naming.

On the other hand, I suggest defining this type the same way as we have it in the `web-react` package `types/shared/disctionaries` and `constants` because the developer will be able to use the value as `Direction.HORIZONTAL`.